Modules/Core/Common/include/itkMacro.h

```diff
@@ -121,12 +121,22 @@ namespace itk
 #if defined(__clang__) && defined(__has_warning)
 #  define ITK_CLANG_PRAGMA_PUSH ITK_PRAGMA(clang diagnostic push)
 #  define ITK_CLANG_PRAGMA_POP ITK_PRAGMA(clang diagnostic pop)
-#  define ITK_CLANG_SUPPRESS_Wzero_as_null_pointer_constant \
-    ITK_PRAGMA(clang diagnostic ignored "-Wzero-as-null-pointer-constant")
+#  if __has_warning("-Wzero-as-null-pointer-constant")
+#    define ITK_CLANG_SUPPRESS_Wzero_as_null_pointer_constant \
+      ITK_PRAGMA(clang diagnostic ignored "-Wzero-as-null-pointer-constant")
+#  else
+#    define ITK_CLANG_SUPPRESS_Wzero_as_null_pointer_constant
+#  endif
+#  if __has_warning("-Wduplicate-enum")
+#    define ITK_CLANG_SUPPRESS_Wduplicate_enum ITK_PRAGMA(clang diagnostic ignored "-Wduplicate-enum")
+#  else
+#    define ITK_CLANG_SUPPRESS_Wduplicate_enum
+#  endif
```

Comment:
I see that the `if __has_warning` is necessary to avoid another warning (_warning: unknown warning group '-W...', ignored [-Wunknown-warning-option]_). Unfortunately it's a bit cumbersome to have 5 lines of boilerplate code to define such a macro (`ITK_CLANG_SUPPRESS_Wduplicate_enum`), for just one warning. Would it be possible to make macro definitions like this more concise? And would it be possible to just specify the string "-Wduplicate-enum" once, instead of twice?  I'm just asking. 🤷 No show stopper.